api/db-cdr/prep-tables/make-prep-survey.py

```diff
@@ -0,0 +1,470 @@
+                    write_row(registered_writer, parent_id_registered, question_code, question_label
+                              , 'QUESTION', None, None, item_flags, 'registered', long_code_to_short_code)
+
+                    if is_numeric_answer:
+                        add_numeric_answer(registered_writer, question_id_registered,
+                                           min_value, max_value, item_flags, 'registered')
+                    else:
+                        add_answers(registered_writer, choices, question_code, question_id_registered
+                                    , None, None, item_flags, 'registered', long_code_to_short_code, answer_suppression)
+
+    # copy local output files and upload it to the bucket
+    storage_client = storage.Client.from_service_account_json('../../sa-key.json')
+    bucket = storage_client.get_bucket('all-of-us-workbench-private-cloudsql')
+    files = [f for f in listdir(home_dir) if isfile(join(home_dir, f))]
+    for f in files:
+        blob = bucket.blob('cb_prep_tables/redcap/output/' + f)
+        blob.upload_from_filename(home_dir + "/" + f)
+
+    # when done remove directory and all files
+    shutil.rmtree(home_dir)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog='make-prep-survey',
+                                     usage='%(prog)s --project <project> '
+                                           '--dataset <dataset> '
+                                           '--date <date>',
+                                     description='Generate 3 csv files')
+    parser.add_argument('--project', type=str, help='project name',
+                        required=True)
+    parser.add_argument('--dataset', type=str, help='dataset name',
+                        required=True)
+    parser.add_argument('--date', type=str, help='date for input file',
+                        required=True)
+    arg = parser.parse_args()
+    return arg.project, arg.dataset, arg.date
+
+
+def read_csv(file):
+    blob = get_blob(file)
+    blob = blob.download_as_string()
+    blob = blob.decode('utf-8')
+    blob = StringIO(blob)  # tranform bytes to string
+    return csv.DictReader(blob)  # then use csv library to read the content
+
+
+def files_exist(date):
+    for name in surveys:
+        file = name + '_' + date + '.csv'
+        blob = get_blob(file)
+        if not blob.exists():
+            raise Exception(file +
+                            " does not exist!")
+
+
+def skip_row(row):
+    return row['Variable / Field Name'].lower() != 'record_id' \
+           and row['Field Type'].lower() != 'descriptive' \
+           and 'please specify' not in row['Field Label'].lower()
+
+
+def parse_row_columns(row):
+    field_annotation = row['Field Annotation']
+    field_annotation_array = []
+    if field_annotation != '  ':
+        [field_annotation_array.append(item.strip()) for item in
+         field_annotation.split(',')]
```

Comment:
Codacy is not happy with your assignment here. Not sure what your trying to do here?